packages/neo-one-client/src/__e2e__/DeveloperClient.test.js

```diff
@@ -143,67 +167,110 @@ async function confirmTransaction(
     );
+    const wallet = await addWallet({ walletName, keystore, networkName });
 
     await developerClient.updateSettings({ secondsPerBlock: 1 });
     await new Promise(resolve => setTimeout(() => resolve(), 15000));
-    const transaction = await setupTransaction(client, master, wallet);
+    const transaction = await setupTransaction({ client, master, wallet });
 
     await confirmTransaction(transaction);
 
-    await checkWalletBalance(walletName, network);
+    await checkWalletBalance({ walletName, networkName });
   });
 
   test('fastForwardOffset', async () => {
-    const network = 'e2e-3';
+    const networkName = 'e2e-3';
+    const secondsPerBlock = 1;
     const offsetSeconds = 1000000000;
 
-    const { developerClient, client } = await setupClients(network);
-    await developerClient.fastForwardOffset(offsetSeconds);
-    await new Promise(resolve => setTimeout(() => resolve(), 15000));
+    const { developerClient, client } = await setupClients(networkName);
 
-    const firstBlock = await client.read(network).getBlock(0);
-    const secondBlock = await client.read(network).getBlock(1);
+    await developerClient.updateSettings({ secondsPerBlock });
+    await new Promise(resolve => setTimeout(() => resolve(), 15000));
 
-    expect(secondBlock.time - firstBlock.time).toBeGreaterThanOrEqual(
-      offsetSeconds,
-    );
+    await developerClient.fastForwardOffset(offsetSeconds);
+    await new Promise(resolve => setTimeout(() => resolve(), 2000));
+
+    const times = await getBlockTimes({ client, networkName });
+
+    let offsetFound = false;
+    for (let i = 0; i < times.length - 2; i += 1) {
+      const offset = times[i + 1] - times[i];
+      if (offset === offsetSeconds + secondsPerBlock) {
+        offsetFound = true;
+        break;
+      }
+    }
+    expect(offsetFound).toEqual(true);
   });
 
   test('fastForwardToTime', async () => {
-    const network = 'e2e-4';
-    const time = utils.nowSeconds() + 1000000000;
+    const networkName = 'e2e-4';
+    const secondsPerBlock = 1;
+    const offset = 1000000000;
 
-    const { developerClient, client } = await setupClients(network);
+    const { developerClient, client } = await setupClients(networkName);
 
-    await developerClient.fastForwardToTime(time);
+    await developerClient.updateSettings({ secondsPerBlock });
     await new Promise(resolve => setTimeout(() => resolve(), 15000));
 
-    const block = await client.read(network).getBlock(1);
-    expect(block.time).toBeGreaterThanOrEqual(time);
-    expect(block.time).toBeLessThanOrEqual(time + 15);
+    const time = utils.nowSeconds() + offset;
+    await developerClient.fastForwardToTime(time);
+    await new Promise(resolve => setTimeout(() => resolve(), 2000));
+
+    const times = await getBlockTimes({ client, networkName });
+
+    let timeFound = false;
+    for (let i = 0; i < times.length - 1; i += 1) {
+      if (times[i] >= time && times[i] <= time + secondsPerBlock) {
```

Comment:
Same comment, seems like this could be flaky no? What happens if there's a delay in processing? Just because it's supposed to produce a block every `secondsPerBlock` doesn't mean it actually will.